### DISCUSSION TITLE
流程编排里面，“代码执行”能否执行更加复杂的python代码

### DISCUSSION THREAD

**Discussion by LuckMonkey2000** [langgenius/dify#3504] [General]
在流程编排里面，我想使用“代码执行”模块执行一个访问数据库的python代码，但是运行会报错,报错信息如下图。dify是否是只支持简单的python脚本，不支持复杂一些的脚本。

![1713208102216](https://github.com/langgenius/dify/assets/25970563/0de3c030-0afa-4681-81a0-65075301b862)

**Comment by crazywoola** [comment]
We support limited packages and operations due to the security reasons. We plan make it more open until we think it's good enough.

**Reply by xixici** [reply]
Well. But in some complex case, we may beyond the domain of security. Is there a method available to solve it?
